prov/gni/src/gnix_cq.c

```diff
@@ -712,12 +704,10 @@ DIRECT_FN int gnix_cq_open(struct fid_domain *domain, struct fi_cq_attr *attr,
 	 */
 	cq_priv->entry_size = format_sizes[cq_priv->attr.format];
 
-
+	fastlock_init(&cq_priv->lock);
 	ret = gnix_cq_set_wait(cq_priv);
 	if (ret)
-		goto err3;
-
-	fastlock_init(&cq_priv->lock);
+		goto err4;
```

Comment:
So err3 was invalid from the get go since the _gnix_ref_get on the cq_priv->domain occurs above err3 in the current code. So that means that the domain would not be ref_put if the gnix_cq_set_wait did a goto err3. 

Since the fastlock_init was below the gnix_cq_set_wait instead of reordering the destroys I removed err3 and put the fastlock_init above the gnix_cq_set_wait. This is to fix a bug in the case that gnix_cq_set_wait failed.
Which would not have been encountered in the past since verify_cq_attr would have bounced out any requests for wait_objects. Now that we can fail in the cq_set_wait_object we need to fix the bug where we might not have called _gnix_ref_put if the gnix_cq_set_wait failed, even though a _gnix_ref_get had already occured.